lib/facebot.js

```diff
@@ -154,11 +155,41 @@ Facebot.prototype.createFBApi = function(credentials){
            });
 }
 
+// loop that will continue to send the is typing indicator to a channel
+// until we hear back they are not typing, or 10 minutes have past
+Facebot.prototype.typingLoop = function(link){
+    if (! link.is_typing || ((Date.now() / 1000) - link.typing_start_time) > 60*10) //if they were typing for more than 10 minutes lets assume we missed the typing end event
+        return;
+    var us = this;
+    this.ws.send(JSON.stringify({ type: 'typing', channel: link.slack_channel }));
+    fbUtil.sleep(3000).then( ()=> us.typingLoop(link));
+
+}
 // Handles any facebook messages received, formats them
 // and sends them through to the linked slack channels
 Facebot.prototype.postFBMessagesToSlack = function(fbmessage){     
-    _.where(this.channelLinks, { fb_thread: fbmessage.threadID.toString() })
-    .forEach(link => {        
+    var threadID = undefined;
+    if (fbmessage.type == "message")
+        threadID = fbmessage.threadID;
+    if (fbmessage.type == "typ")//not sure why typing messages appear on a different thread but they do so we need to use from...
+        threadID = fbmessage.from;
+    if (! threadID)
+        return;
+  
+    _.where(this.channelLinks, { fb_thread: threadID.toString() })
+    .forEach(link => {
+        if (fbmessage.type == "typ"){
+            if (fbmessage.isTyping == link.is_typing)
+                return;
+            link.is_typing = fbmessage.isTyping;
+            if (link.is_typing){
+                link.typing_start_time = Date.now() / 1000;
+                this.typingLoop(link);
+            }
+        }
+        if (fbmessage.type != "message")
```

Comment:
I think it would be nicer if this were:
```
if(fbmessage.type === "message"){
```

As its more consistent with the code above, and makes it a bit easier to follow